applications/asset_tracker/src/main.c

```diff
@@ -219,6 +221,36 @@ void error_handler(enum error_type err_type, int err_code)
 #endif /* CONFIG_DEBUG */
 }
 
+void z_SysFatalErrorHandler(unsigned int reason,
+			    const NANO_ESF *pEsf)
+{
+	ARG_UNUSED(pEsf);
+
+#if !defined(CONFIG_SIMPLE_FATAL_ERROR_HANDLER)
+#if defined(CONFIG_STACK_SENTINEL)
+	if (reason == _NANO_ERR_STACK_CHK_FAIL) {
+		goto error_handler;
+	}
+#endif
+	if (reason == _NANO_ERR_KERNEL_PANIC) {
+		goto error_handler;
+	}
+
```

Comment:
There are more reasons you can list here, check Zephyr's default fatal error handler for ARM

Reply:
that's up to you of course

Reply:
I believe the main reasoning here is that we would like to mitigate the amount of code added to main whilst still maintaining a close-to-equal functionality. I can add them if you believe it should be there, but if not I'll let it be.